python/python/tests/test_dataset.py

```diff
@@ -1545,3 +1545,15 @@ def test_migrate_manifest(tmp_path: Path):
     ds = lance.write_dataset(table, tmp_path)
     # We shouldn't need a migration for a brand new dataset.
     assert not manifest_needs_migration(ds)
+
+
+def test_v2_dataset(tmp_path: Path):
+    table = pa.table({"a": range(100), "b": range(100)})
+    # max_rows_per_group is ignored by the v2 writer so we specify here to confirm
+    # we are in fact using v2 files
```

Comment:
🤣  I hope in the future there is a more straight forward way to check this.

Reply:
Ok, I added the file versions & column indices from the data file into `format_fragment` and now I use that to verify this info.